### ISSUE TITLE
Questions about model complexity

### ISSUE THREAD

**Issue by changshuowang** [kaist-avelab/PointStack#1]
Dear author, thank you for your work, I would like to express some of my views on the issue of PointStack complexity.

I tested the complexity of this model and the results are as follows:

############################################################
<img width="428" alt="1" src="https://user-images.githubusercontent.com/49676110/170892025-232da95f-7df5-4b26-8ae9-2a12de74e713.png">

<img width="603" alt="2" src="https://user-images.githubusercontent.com/49676110/170892026-b6695d2d-64c4-4d8d-8c58-82cb9bef3d0e.png">
##########################################################################

**Question1:**
        The complexity of this PointStack on modelNet40 is shocking, when the number of samples is 2, the flops is 38.7G, and when the number of samples is 3, the flops is 58.1G, so it can be inferred that when the model infers one sample, the flops is about **19.4G** , the model parameter is **100.3M**. This model has **fifty or sixty times as many parameters as most models**. 
        Therefore, what kind of accuracy can this model achieve at the same level of complexity as most models?

**Question2:**
        PointStack has achieved superior performance on ScanobjectNN and ShapeNetPart, but only reached **93.3%** on modelNet40. **Is this caused by the excessive number of parameters of PointStack and overfitting on modelnet40?**

**Comment by ktirta** [comment]
@changshuowang Thanks for your interest in our work!

The number of parameters of PointStack is larger compared to existing models because we use learnable queries for mha operation in the learnable pooling. As shown in your first image, most of the parameters are for the mha operations. You can refer to Subsection 4.4 in the paper (Table 2) for a general idea of PointStack's performance with lower numbers of trainable parameters. Referring to module names in your first picture, removing multi-resolution LP should eliminate the global_mha parameters, and removing single-resolution LP should eliminate the mha_list and fc_pre_mha_list parameters.

For the second question, we discuss the lower performance on ModelNet40 in Subsection 4.6 of the paper. Experimental results indicate that PointStack requires a larger amount of training samples compared to existing models, which could be a result of the higher number of trainable parameters.

**Comment by changshuowang** [comment]
@ktirta Thanks to the author for your reply.
         your idea is novel, but the complexity of the model is relatively high, so we cannot know **whether it is the network structure design or the large number of parameters that makes the network achieve superior performance**? 

        Through your reply, I think readers generally pay attention to the overall complexity of the model, and mha is only the reason for the large number of parameters in the model. **Therefore, my suggestion is to reduce the network complexity by simplifying the mha module,** because I see that the code is modified based on PointMLP. The parameter amount of PointMLP is 12.6M, and yours has reached 100.4M. During the review process, if the reviewers see that your network complexity is so large that it will be your disadvantage. 

        And it can be seen from Table 2, what is the amount of model parameters after removing multi-resolution LP and single-resolution LP? If it is about 2M, then the model complexity is from 2M to 100.4M, **and the accuracy of the model on scanobjectNN is only increased by 1%. Is this improvement too small?**

For the second question, through your reply, **I think your model is overfitting on ModelNet40**, so my suggestion is to reduce the complexity of the model and see if the effect has further improvement.

Maybe my understanding is wrong, this is just my suggestion, I hope to see your paper accepted as soon as possible.

**Comment by ktirta** [comment]
@changshuowang Thanks for the interesting insights!

Indeed, model complexity may obscure whether the network structure is beneficial or not. However, I would say that the original PointStack benefits from both network structure and more parameters. First, with only multi-resolution feature structure (i.e. similar number of parameters to baseline), we already see classification improvement. Second, adding more parameters with the learnable pooling structure further improves the classification performance. This is contrary to adding more parameters by simply making the network deeper, which actually deteriorates the classification performance as seen in the PointMLP paper.

On the other hand, we just ran an additional experiment with a significantly reduced number of parameters (~14.6M). We are delighted to observe that the classification performances of this single run are 86.8 (OA) and 86.1 (mAcc) on ScanObjectNN, which is within 1 standard deviation of the reported result in the paper. We will give more updates after running more experiments.

Once again, thank you for the suggestions!

**Comment by ktirta** [comment]
We have updated the links and results for ScanObjectNN and ShapeNetPart with the new configs.
